app/utils/media.js

```diff
@@ -0,0 +1,79 @@
+const axios = require('axios');
+const config = require('../../config');
+
+const getClosestSize = (sizes, width) => {
+    const closest = { 'position': 0, 'diff': null };
+    sizes.forEach((size, index) => {
+        const diff = Math.abs(width - size.width);
+        if (closest.diff === null || closest.diff > diff) {
+            closest.diff = diff
+            closest.position = index
+        }
+    });
+    return sizes[closest.position];
+}
+
+const generateUrlFromIIIFSize_info = async (size_info_url, width) => {
+    const { data } = await axios.get(size_info_url);
+    const size = getClosestSize(data.sizes, width)
+    return data.id + "/full/" + size.width + "," + size.height + "/0/default.jpg";
+}
+
+const getImageMediaUrlFromIIIFSize_info = async (media, size_info_url, width) => {
+    media.image_url = await generateUrlFromIIIFSize_info(size_info_url, width)
+    return;
+}
+
+const isIIIFLevel0 = (iiif_data) => {
+  return iiif_data && iiif_data.size_info;
+}
+
+exports.generateImageUrl = async (media, image_id, collection_id, iiif_data, region, image_width, image_height, iiifLevel0_width ) => {
```

Comment:
maybe we need to make the naming clearer. I would suggest `setImageUrl` and `setIIIFLevel0BannerUrl` for method that set a value on the object directly and `generate` for the one that return a value